src/main/scala/Joint.scala

```diff
@@ -32,7 +32,74 @@ object Joint {
     val n = DenseVector[Double](normalVec._1, normalVec._2, normalVec._3)
     -(n dot w)/linalg.norm(n)
   }
+
+  /**
+   * Find the vector indicating dip direction of the joint plane. Global positive x-axis points North and z-axis oriented
+   * with positive upward. Positive y-axis will point west based on this orientation.
+   * @param normalVec Normal vector to the joint plane
+   * @return Dip direction of the plane, indicating direction of greatest increase in z. Return as vector (a, b, 0)
+   */
+  private def dipDirVector(normalVec: (Double, Double, Double)) = {
+    // Dip direction is in opposite direction of gradient indicating greatest increase in z.
+    if ((math.abs(normalVec._3) > Joint.EPSILON) && (math.abs(math.abs(normalVec._3) - 1.0) > Joint.EPSILON)) {
+      DenseVector[Double](normalVec._1 / normalVec._3, normalVec._2 / normalVec._3, 0.0)
+    } else if (math.abs(normalVec._3) < Joint.EPSILON) {
+      // Joint is vertical, assigns non-zero z component that will be caught is dipDir function
+      DenseVector[Double](0.0, 0.0, -1.0)
+    } else {
+      // Joint is horizontal, dip direction arbitrarily assigned to 90 degrees so that strike is 0 degrees
+      DenseVector[Double](0.0, -1.0, 0.0)
+    }
+  }
+
+  /**
+   * Finds the dip direction of the input joint as an azimuth. Global positive x-axis points North.
+   * @param normalVec Normal vector to the joint plane
+   * @return Dip direction as an azimuth in radians
+   */
+  private def dipDir(normalVec: (Double, Double, Double)) = {
+    val dipVector = Joint.dipDirVector(normalVec)
+    val xAxis = DenseVector[Double](1.0, 0.0, 0.0)
+    if (dipVector(2) != -1.0) {
```

Comment:
It's usually a good idea to avoid checking doubles for exact equality. You may want to change this to `math.abs(dipVector(2) + 1) < Joint.EPSILON`.